java/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/PartitionReceiver.java

```diff
@@ -312,6 +328,10 @@ public void setReceiveHandler(final PartitionReceiveHandler receiveHandler)
 			}
 			else
 			{
+				if (this.isOnReceivePumpRunning)
+					throw new IllegalArgumentException(
+							"Unexpected value for parameter 'receiveHandler'. Close the existing running pump and then set the new handler");
```

Comment:
>Unexpected value for parameter 'receiveHandler' [](start = 8, length = 47)

we should just say only one receiveHandler can be registered with one receiver (pump running means we already set it once).